src/integrations/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -109,15 +123,47 @@ public void Init(HttpApplication application)
 			}
 			catch (Exception ex)
 			{
-				const string linePrefix = "Elastic APM .NET Agent: ";
-				Trace.WriteLine($"{linePrefix}[CRITICAL] Exception thrown by {nameof(ElasticApmModule)}.{nameof(Init)}."
-					+ Environment.NewLine + linePrefix + $"+-> Exception: {ex.GetType().FullName}: {ex.Message}"
-					+ Environment.NewLine + TextUtils.PrefixEveryLine(ex.StackTrace, linePrefix + " ".Repeat(4))
-				);
+				const string message = $"Exception thrown by {nameof(ElasticApmModule)}.{nameof(Init)}.";
+
+				// Prefer logging to the IApmLogger if that was able to be assigned.
+				if (_logger is not null)
+				{
+					_logger.Critical()?.LogException(ex, message);
+					return;
+				}
+
+				// Fallback to writing to any trace listeners.
+				WriteTraceLog(ex, message);
+			}
+		}
+
+		private void WriteTraceLog(Exception ex, string message)
+		{
+			const string linePrefix = "Elastic APM .NET Agent: ";
+
+			Trace.WriteLine($"{linePrefix}[CRITICAL] {message}"
+				+ Environment.NewLine + linePrefix + $"+-> Exception: {ex.GetType().FullName}: {ex.Message}"
+				+ Environment.NewLine + TextUtils.PrefixEveryLine(ex.StackTrace, linePrefix + " ".Repeat(4)));
+		}
+
+		private void AssignScopedLogger()
+		{
+			try
+			{
+				var logger = AgentDependencies.Logger ?? FullFrameworkDefaultImplementations.CreateDefaultLogger(null);
+				_logger = logger.Scoped(_dbgInstanceName);
+			}
+			catch (Exception ex)
+			{
+				const string message = $"Exception thrown by {nameof(ElasticApmModule)}.{nameof(AssignScopedLogger)}.";
```

Comment:
Q? Why would this ever throw an exception? Might be good to note this too. 


Having to document this is less than ideal but if necessary c'est la vie.
https://github.com/elastic/apm-agent-dotnet/pull/2302/files

Reply:
I was over-cautious here as everything was guarded by the original try/catch. However, having re-checked the code path, I can't see a scenario where an exception could occur when creating the logger. I guess we could assume that at least the logger creation will succeed and avoid the fallback. The only other place where we fall back to `Trace.WriteLine` is in `LoggingExtensions`.

Reply:
Cool. I'll amend this PR, and then the docs one can be closed out.